devel/export-map.lua

```diff
@@ -0,0 +1,332 @@
+-- Export fortress map tile data to a JSON file
+-- based on export-map.lua by mikerenfro:
+-- https://github.com/mikerenfro/df-map-export/blob/main/export-map.lua
+-- redux version by timothymtorres
+
+local tm = require('tile-material')
+local utils = require('utils')
+local json = require('json')
+local argparse = require('argparse')
+
+local include_underworld_z = false
+local underworld_z
+local evilness
+
+-- the layer of the underworld
+for _, feature in ipairs(df.global.world.features.map_features) do
+    if feature:getType() == df.feature_type.underworld_from_layer then
+        underworld_z = feature.layer
+    end
+end
+
+-- right now the only tile_liquids are Water and Magma
+local liquid_list = {}
+for id, liquid in ipairs(df.tile_liquid) do
+    liquid_list[id] = string.upper(liquid)
+end
+
+-- copied from agitation-rebalance.lua
+-- check only one tile at the center of the map at ground lvl
+-- (this ignores different biomes on the edges of the map)
+local function get_evilness()
+    -- check around ground level
+
+    local lvls_above
+    lvls_above = df.global.world.worldgen.worldgen_parms.levels_above_ground
+    local ground_z = (df.global.world.map.z_count - 2) - lvls_above
+    local xmax, ymax = dfhack.maps.getTileSize()
+    local center_x, center_y = math.floor(xmax/2), math.floor(ymax/2)
+    local rgnX, rgnY = dfhack.maps.getTileBiomeRgn(center_x, center_y, ground_z)
+    local biome = dfhack.maps.getRegionBiome(rgnX, rgnY)
+
+    return biome and biome.evilness or 0
+end
+
+local function classify_tile(options, x, y, z)
+    -- The last z-levels of hell shrink their x/y size unexpectedly! (ಠ_ಠ)
+    -- if your map is 190x190, the last hell z-levels are gonna be like 90x90
```

Comment:
not all the time. this is only likely to happen if your map happens to cross a region boundary and different regions are different depths.

Reply:
Add those extra details via https://github.com/DFHack/scripts/pull/1329/commits/019527f8ee79cd1d5b71b1c91d40d0252aba3ec1